File: lib/data/non/input/touch.rb
module Touch
    java_import 'com.badlogic.gdx.Gdx'

    def self.x(pointer = 0)
        Gdx.input.getX(pointer)
    end
    
    def self.y(pointer = 0)
        Gdx.input.getY(pointer)
    end
    
    def self.down(pointer = 0)
        Gdx.input.isTouched(pointer)
    end
end

File: lib/data/non/lua/input/touch.lua
non.touch = {}

local Gdx = NON_GDX

function non.touch.x(pointer)
  if pointer == nil then pointer = 0 end
  return Gdx.input:getX(pointer)
end
  
function non.touch.y(pointer)
  if pointer == nil then pointer = 0 end
  return Gdx.input:getY(pointer)
end
  
function non.touch.down(pointer)
  if pointer == nil then pointer = 0 end
  return Gdx.input:isTouched(pointer)
end